### PR TITLE
Fail the work item when pre and post commands fail.

### PR DESCRIPTION
**Pull request by alexperovich**
https://github.com/dotnet/arcade/issues/6919

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/Sdk/SendHelixJob.cs [hunk 1]
```diff
@@ -388,19 +388,29 @@ private IJobDefinition AddWorkItem(IJobDefinition def, ITaskItem workItem)
...
         private IEnumerable<string> GetCommands(ITaskItem workItem, string workItemCommand)
         {
+            string MakeCommand(string command)
+            {
+                if (IsPosixShell)
+                {
+                    return $"{{ {command} }} || exit $?";
+                }
+
+                return $"{command} || exit /b";
```
Comment (ChadNedzlek) [inline src/Microsoft.DotNet.Helix/Sdk/SendHelixJob.cs:398]: Exit b... with what exit code? Does that return the current exit code?

**Review by ChadNedzlek — APPROVED** [review]

**Review by ChadNedzlek — APPROVED** [review]

**Review by MattGal — CHANGES_REQUESTED** [review]
This is doing it to every pre/post command.  Why not just do it to the one inserted by Azure Devops reporter by sticking the || exit bits only in that case?

File: src/Microsoft.DotNet.Helix/Sdk/SendHelixJob.cs (hunk 1 above)
Comment (alexperovich) [reply src/Microsoft.DotNet.Helix/Sdk/SendHelixJob.cs:398]: yep, it preserves the %ERRORLEVEL%, I could explicitly put !ERRORLEVEL! there, but then I have to also enable delayed expansion

Comment (ChadNedzlek) [reply src/Microsoft.DotNet.Helix/Sdk/SendHelixJob.cs:398]: Nope, that's fine. Batch is 95% magic anyway, so I'm totally fine with this.

**Pull request closed without merging**